types/types.go

```diff
@@ -181,9 +181,9 @@ type NodeConfig struct {
 	Memory string  `json:"memory,omitempty"`
 
 	// Extra node parameters
-	Extras  *Extras    `json:"extras,omitempty"`
-	WaitFor []string   `json:"wait-for,omitempty"`
-	DNS     *DNSConfig `json:"dns,omitempty"`
+	Extras  *Extras                     `json:"extras,omitempty"`
+	WaitFor map[WaitForPhase][]*WaitFor `json:"wait-for,omitempty"`
```

Comment:
@steiler  this is cool

I wonder how do you feel about this layout:


```yaml
name: linux

topology:
  kinds:
    linux:
      image: alpine
  nodes:
    l1:
      kind: linux
    l2:
      kind: linux
      stage:
        configure:
          wait-for:
            - node: l1
              status: created
    l3:
      kind: linux
      image: mavatest
      stage:
        configure:
          wait-for:
            - node: l2
              status: configured
    l4:
      kind: linux
      stage:
        create:
          wait-for:
            - node: l3
              status: healthy
        configure:
          wait-for:
            - node: l1
                status: configured
```

The idea here is that nodes have a `stage` that denotes a certain stage a node goes through the deployment process.

Each stage has a `wait-for` that works like you cleverly did here.
But it also allows us to parameterized stages by introducing other nested structs.
Like

```yaml
    l2:
      kind: linux
      stage:
        configure:
          skip: true
```

Reply:
Uhhh I see nice idea...